### DISCUSSION TITLE
VPP that prints QP values of AuxVariables and coordinates?

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#22779] [Q&A General] [answered]
I'd like to get text output from a MOOSE run that shows my AuxVariable values at all quadrature points along with the coordinates. Could be for all times or each time step in a separate file or whatever. I don't see any VectorPostprocessors that do this, but [MaterialVectorPostprocessor](https://mooseframework.inl.gov/source/vectorpostprocessors/MaterialVectorPostprocessor.html) is close if I make a ParsedMaterial from my AuxVariable, but a) it is only for specified elements and b) does not print the coordinates. Could that VPP be modified to do this or is there a better way? Through Paraview perhaps?

**Comment by jessecarterMOOSE** [comment, marked as answer]
Oh I forgot about [Sampler1DReal](https://mooseframework.inl.gov/modules/thermal_hydraulics/source/vectorpostprocessors/Sampler1DReal.html). Why is that in the Thermal Hydraulics module?

**Reply by jessecarterMOOSE** [reply]
See https://github.com/idaholab/moose/discussions/22100

**Reply by GiudGiud** [reply]
no idea. 
Another sampler for variables is the ElementValueSampler, but that does not Qps just centroids

**Reply by jessecarterMOOSE** [reply]
Would it be easy to add to or inherit from `ElementValueSampler` to have it work at quadrature points?

**Reply by GiudGiud** [reply]
I d expect it to be easier from the MaterialVPP but ElementValueSampler could also be an option

**Reply by jessecarterMOOSE** [reply]
ElementValueSampler seems closer since it already prints coordinates and runs on all elements. Would it be difficult to add a qp loop in there? I'm not sure how that `Sampler` class works.
